internal/controlplane/handlers_githubwebhooks.go

```diff
@@ -145,42 +334,64 @@ func (s *Server) HandleGitHubAppWebhook() http.HandlerFunc {
 		}
 
 		wes.Typ = github.WebHookType(r)
-		if wes.Typ == "ping" {
-			logPingReceivedEvent(r.Context(), rawWBPayload)
-			wes.Error = false
-			return
-		}
 
 		m := message.NewMessage(uuid.New().String(), nil)
 		m.Metadata.Set(events.ProviderDeliveryIdKey, github.DeliveryID(r))
-		m.Metadata.Set(events.ProviderSourceKey, "https://api.github.com/") // TODO: handle other sources
+		// TODO: handle other sources
+		m.Metadata.Set(events.ProviderSourceKey, "https://api.github.com/")
 		m.Metadata.Set(events.GithubWebhookEventTypeKey, wes.Typ)
 
-		ctx := r.Context()
 		l := zerolog.Ctx(ctx).With().
 			Str("webhook-event-type", m.Metadata[events.GithubWebhookEventTypeKey]).
 			Str("providertype", m.Metadata[events.ProviderTypeKey]).
 			Str("upstream-delivery-id", m.Metadata[events.ProviderDeliveryIdKey]).
 			Logger()
+		ctx = l.WithContext(ctx)
+
+		wes.Accepted = true
+		var res *processingResult
+		var processingErr error
+
+		switch github.WebHookType(r) {
+		case "ping":
+			// For ping events, we do not set wes.Accepted
+			// to true because they're not relevant
+			// business events.
+			wes.Accepted = false
+			wes.Error = false
+			s.processPingEvent(ctx, rawWBPayload)
+		case "installation":
+			res, processingErr = s.processInstallationAppEvent(ctx, rawWBPayload)
+		default:
+			l.Info().Msgf("webhook event %s not handled", wes.Typ)
+		}
 
-		if err := s.parseGithubAppEventForProcessing(rawWBPayload, m); err != nil {
-			wes = handleParseError(wes.Typ, err)
+		if processingErr != nil {
+			wes = handleParseError(wes.Typ, processingErr)
 			if wes.Error {
 				w.WriteHeader(http.StatusInternalServerError)
-			} else {
-				w.WriteHeader(http.StatusOK)
+				return
 			}
+
+			w.WriteHeader(http.StatusOK)
 			return
 		}
 
-		wes.Accepted = true
-		l.Info().Str("message-id", m.UUID).Msg("publishing event for execution")
+		if res != nil {
+			l.Info().Str("message-id", m.UUID).Msg("publishing event for execution")
+			if err := res.iiw.ToMessage(m); err != nil {
```

Comment:
would it make sense to double-check that iiw is non-nil here in case we refactor the code later to handle more types?

Reply:
It might be the case that we want to return an interface and switch-case on its actual type.
Which way would you prefer?

Reply:
I don't really have a preference other than we don't panic :-)